### ISSUE TITLE
'train.py' and 'train_parallel.py' training comparison

### ISSUE THREAD

**Issue by cysin** [deepinsight/insightface#576]
It's a feedback rather than an issue report. I did a simple test with 'train.py' and 'train_parallel.py' and observed that 'train_parallel.py' got a relatively lower validation accuracy compared to 'train.py'

Data:
`~0.9M identities, >20M images`

env1: 'train.py':
`V100 32GB x 4, batch: 128, ~310 samples/second`

env2: 'train_parallel'
`2080Ti 11GB x4, batch:50 (got OOM if set to 64) ~450 samples/second`

So with same samples processed(about more than one epoch, lr=0.1), the validation results are:

env1:
```
infer time 13.888804
[lfw][2000]XNorm: 22.394393
[lfw][2000]Accuracy-Flip: 0.98817+-0.00398
testing verification..
(14000, 512)
infer time 16.613949
[cfp_fp][2000]XNorm: 18.647367
[cfp_fp][2000]Accuracy-Flip: 0.85086+-0.02107
testing verification..
(14000, 512)
infer time 16.212121
[cfp_ff][2000]XNorm: 21.757638
[cfp_ff][2000]Accuracy-Flip: 0.98071+-0.00614
testing verification..
(10000, 512)
infer time 11.568479
[vgg2_fp][2000]XNorm: 19.112889
[vgg2_fp][2000]Accuracy-Flip: 0.87240+-0.01637
testing verification..
(12000, 512)
infer time 12.080508
[agedb_30][2000]XNorm: 21.364253
[agedb_30][2000]Accuracy-Flip: 0.90033+-0.01751
```

env2:
```
infer time 8.597218
[lfw][190000]XNorm: 22.441565
[lfw][190000]Accuracy-Flip: 0.98000+-0.00711
testing verification..
(14000, 512)
infer time 10.06084
[cfp_fp][190000]XNorm: 19.034261
[cfp_fp][190000]Accuracy-Flip: 0.81586+-0.01070
testing verification..
(14000, 512)
infer time 9.790307
[cfp_ff][190000]XNorm: 21.557386
[cfp_ff][190000]Accuracy-Flip: 0.96214+-0.00724
testing verification..
(10000, 512)
infer time 6.909437
[vgg2_fp][190000]XNorm: 19.837395
[vgg2_fp][190000]Accuracy-Flip: 0.82740+-0.01302
testing verification..
(12000, 512)
infer time 8.352047
[agedb_30][190000]XNorm: 21.250625
[agedb_30][190000]Accuracy-Flip: 0.84017+-0.01907
[190000]Accuracy-Highest: 0.85383
```

**Comment by nttstar** [comment]
Make sure you have the same batch-size

**Comment by cysin** [comment]
Thanks for the reply!

**Comment by clhne** [comment]
@nttstar 
When run train_parall.py, the callback's acc and lossvalue is nan, which can cause this?
Thanks.

**Comment by EtheneXiang** [comment]
@nttstar  
@cysin 
hello,when i run the train_parall.py,i got some errors,:
![image](https://user-images.githubusercontent.com/49241737/59846688-19923a80-9393-11e9-8d08-6ad2c80d91b9.png)
while i run train.py is ok:
export MXNET_CPU_WORKER_NTHREADS=24
export MXNET_CUDNN_AUTOTUNE_DEFAULT=0
export MXNET_ENGINE_TYPE=ThreadedEnginePerDevice
CUDA_VISIBLE_DEVICES='0,1,2,3,4,5,6,7' python2 -u train.py --dataset xxxx--network r100 --loss arcface

you said "Replace train.py with train_parall.py in following examples if you want to use parallel acceleration." in here : https://github.com/deepinsight/insightface/tree/master/recognition
so i tried,but error,.

**Comment by clhne** [comment]
@EtheneXiang 
Check config.py, in line 178, if 
default.pretrained = ' path_of_your_pretrained_model', 
ie. isn't NULL, you should modify the train_parall.py in line 186 to:
if len(args.pretrained)!=0:

**Comment by EtheneXiang** [comment]
yes,you siad it,thx,it works well@clhne 
how about this warning ,what it means:
only 24 out of 56 GPU pairs are enabled direct access. It may affect the performance.
You can set MXNET_ENABLE_GPU_P2P=0 to turn it off
[16:57:18] src/kvstore/././comm.h:754: .vvv....
[16:57:18] src/kvstore/././comm.h:754: v.vv....
[16:57:18] src/kvstore/././comm.h:754: vv.v....
[16:57:18] src/kvstore/././comm.h:754: vvv.....
[16:57:18] src/kvstore/././comm.h:754: .....vvv
[16:57:18] src/kvstore/././comm.h:754: ....v.vv
[16:57:18] src/kvstore/././comm.h:754: ....vv.v
[16:57:18] src/kvstore/././comm.h:754: ....vvv.

**Comment by chinmayjog13** [comment]
@EtheneXiang did you find out what that warning means?
